yjit/src/codegen.rs

```diff
@@ -6774,6 +6779,17 @@ fn iseq_get_return_value(iseq: IseqPtr, captured_opnd: Option<Opnd>) -> Option<I
         return None;
     }
     match first_insn {
+        YARVINSN_getlocal_WC_0  => {
+            // Only accept simple positional only cases for both the caller and the callee.
+            // Reject block ISEQs to avoid autosplat and other block parameter complications.
```

Comment:
Rejecting the `captured_opnd.is_some()` case only impacts `yield`. Is this not a problem for `block.call`?

Reply:
It is a problem for `block.call`, but we essentially do a manual fallback with rb_optimized_call() for those cases 😅 All the block setup code check `captured_opnd `.